pkg-manager/plugin-commands-installation/src/installDeps.ts

```diff
@@ -157,6 +162,15 @@ when running add/update with the --workspace option')
       ? await findWorkspacePackages(opts.workspaceDir, { ...opts, patterns: opts.workspacePackagePatterns })
       : []
   )
+  if (opts.allProjects && opts.workspaceDir) {
+    await updatePackagesList({
+      allProjects: opts.allProjects,
+      catalogs: opts.catalogs,
+      filtered: (opts.filter?.length ?? 0) > 0,
+      lastValidatedTimestamp: Date.now(),
+      workspaceDir: opts.workspaceDir,
+    })
+  }
```

Comment:
how can you set lastValidatedTimestamp at this point if installation wasn't yet performed? This file should be written at the end of the function. Maybe even moved to `@pnpm/core`.

Reply:
IIRC, we don't need `lastValidatedTimestamp` to to newer than any of the lockfiles, we only need it to be newer than the manifest files.

Reply:
I think both. But in any case, at this point installation didn't finish and you cannot know in advance if it will succeed, right? They are validated only if installation succeeded.

Reply:
I had hoped that using `.modules.yaml` to store packages list would have solved this problem, but now that it proves to not make sense, I wonder where should this code be moved to. Is there a place in the code where all successful install will execute?

Reply:
Unfortunately, the fact that we have non-shared lockfile workspaces complicates things. We can't do it inside `@pnpm/core`. We should save the file somewhere in `@pnpm/plugin-commands-installation` after `mutateModules` finishes.

Reply:
I initially wanted to add it to `recursive` directly, but it has too many `return` statement, so I think wrapper is nicer.